app/models/user.rb

```diff
@@ -38,12 +40,47 @@ def remember
     update_attribute(:remember_digest, User.digest(remember_token))
   end
 
-  def authenticated?(remember_token)
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password?(token)
   end
 
   def forget
     update_attribute(:remember_digest, nil)
   end
+
+  def activate
+    update_columns(activated: true,
+                   activated_at: Time.zone.now)
+  end
+
+  def password_reset_expired?
+    reset_sent_at < 2.hours.ago
+  end
+
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_columns(reset_digest: User.digest(reset_token),
+                     reset_sent_at: Time.zone.now)
```

Comment:
__[rubocop]__
* Align the elements of a hash literal if they span more than one line.

<!-- Comment by [SideCI](https://sideci.com) -->